presto-main/src/main/java/com/facebook/presto/sql/planner/plan/GroupIdNode.java

```diff
@@ -81,7 +81,9 @@ public GroupIdNode(
     {
         super(sourceLocation, id, statsEquivalentPlanNode);
         this.source = requireNonNull(source);
-        this.groupingSets = listOfListsCopy(requireNonNull(groupingSets, "groupingSets is null"));
+        checkArgument(requireNonNull(groupingSets, "groupingSets is null").size() > 1,
```

Comment:
I noticed that we only create a GroupId node iff the size of the grouping sets is >1. So adding this invariant here, made sense. No tests failed, so IMO this should be added cc: @rschlussel